ts/Stock/Navigator/StandaloneNavigator.ts

```diff
@@ -0,0 +1,401 @@
+/* *
+ *
+ *  (c) 2010-2024 Mateusz Bernacik
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type { StandaloneNavigatorOptions } from './NavigatorOptions';
+import type { SeriesOptions } from '../../Core/Series/SeriesOptions';
+import type { Options } from '../../Core/Options';
+import Chart from '../../Core/Chart/Chart.js';
+import Navigator, { SetRangeEvent } from './Navigator.js';
+import G from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+import Axis from '../../Core/Axis/Axis.js';
+import standaloneNavigatorDefaults from './StandaloneNavigatorDefaults.js';
+const {
+    merge,
+    addEvent,
+    fireEvent,
+    pick
+} = U;
+
+declare module '../../Core/GlobalsLike.d.ts' {
+    interface GlobalsLike {
+        navigators: Array<StandaloneNavigator>;
+    }
+}
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The StandaloneNavigator class. The StandaloneNavigator class allows for
+ * creating a standalone navigator component that synchronizes the extremes
+ * across multiple bound charts.
+ *
+ * @class
+ * @name Highcharts.StandaloneNavigator
+ *
+ * @param {string|Highcharts.HTMLDOMElement} [renderTo]
+ * The DOM element to render to, or its id.
+ *
+ * @param {StandaloneNavigatorOptions} userOptions
+ * The standalone navigator options.
+ */
+class StandaloneNavigator {
+
+    public eventsToUnbind: Array<Function> = [];
+    public navigator: Navigator;
+    public boundAxes: Array<Axis> = [];
+    public chartOptions: Partial<Options>;
+    public userOptions: StandaloneNavigatorOptions;
+
+    /* *
+     *
+     *  Static Functions
+     *
+     * */
+
+    /**
+     * Factory function for standalone navigator.
+     *
+     * @function Highcharts.navigator
+     *
+     * @param {string|Highcharts.HTMLDOMElement} [renderTo]
+     * The DOM element to render to, or its id.
+     *
+     * @param {StandaloneNavigatorOptions} options
+     * The standalone navigator options with chart-like structure.
+     *
+     * Returns the navigator object.
+     */
+    public static navigator(
+        renderTo: (string|globalThis.HTMLElement),
+        options: StandaloneNavigatorOptions
+    ): StandaloneNavigator {
+        const nav = new StandaloneNavigator(renderTo, options);
+
+        if (!G.navigators) {
+            G.navigators = [nav];
+        } else {
+            G.navigators.push(nav);
+        }
+
+        return nav;
+    }
+
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+
+    constructor(
+        element: (string | globalThis.HTMLElement),
+        userOptions: StandaloneNavigatorOptions
+    ) {
+        this.userOptions = userOptions;
+        this.chartOptions = merge(
+            (G as any).getOptions(),
```

Comment:
Hm.. could you try using Defaults instead of Globals? Like this:
```
import D from '../Defaults.js';
const { getOptions } = D;
```

Perhaps we can get rid of this as any casting

Reply:
I've tried it but still results in a problem with types.